libs/auth/src/certs.ts

```diff
@@ -370,20 +393,39 @@ export function constructCardCertSubjectWithoutJurisdictionAndCardType(): string
   return certSubject;
 }
 
+/**
+ * Parses the provided cert and returns machine details. Throws an error if the cert doesn't follow
+ * VotingWorks's machine cert format.
+ */
+export async function parseMachineDetailsFromCert(
+  cert: Buffer
+): Promise<MachineDetails> {
+  const certDetails = await parseCert(cert);
+  assert(certDetails.component !== 'card', 'Cert is not a machine cert');
```

Comment:
is this the right assert, or can we future proof with `component==='machine'` (or maybe we don't have a component for machines?)

Reply:
`component` is otherwise the type of machine, like `mark-scan`, `admin`, etc.

Reply:
Yeah Brian covered what I was going to say. There is no catch-all `machine` component, just component types for every machine.

`component !== card` does narrow the type down to the other components, all of which are machines. I could alternatively delineate every machine, e.g., `component === 'admin' || component === 'central-scan' || ...`.

Regardless, on the next line, we run:

```
const { component, machineId } = certDetails;
```

Which only compiles successfully if `certDetails` has been narrowed down to a machine cert details object (card certs don't have a machine ID). So the current code is safe / future-proof as is